lib/utils/npm-utils.js

```diff
@@ -79,8 +96,31 @@ function fetchPeerDependencies(packageName) {
 
     if (error && error.code === "ENOENT") {
 
-        // TODO: should throw an error instead of returning null
-        return null;
+        // Fallback to using the npm registry API directly when npm is not available.
```

Comment:
I think we should first check if npm is available via `npm --version` command and if it's available use `npm show --json` otherwise use fetch API.  As per the current implementation, it will only use fetch if the error code is `ENOENT`.

Reply:
@snitin315 your feedback is wanted here. :)

Reply:
I think it's fine if the error code is same `ENOENT`.